src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsremandandsentencingapi/service/RecallService.kt

```diff
@@ -4,23 +4,40 @@ import jakarta.persistence.EntityNotFoundException
 import org.springframework.stereotype.Service
 import org.springframework.transaction.annotation.Transactional
 import uk.gov.justice.digital.hmpps.hmppsremandandsentencingapi.controller.dto.CreateRecall
-import uk.gov.justice.digital.hmpps.hmppsremandandsentencingapi.controller.dto.CreateRecallResponse
 import uk.gov.justice.digital.hmpps.hmppsremandandsentencingapi.controller.dto.Recall
+import uk.gov.justice.digital.hmpps.hmppsremandandsentencingapi.controller.dto.SaveRecallResponse
 import uk.gov.justice.digital.hmpps.hmppsremandandsentencingapi.jpa.entity.RecallEntity
 import uk.gov.justice.digital.hmpps.hmppsremandandsentencingapi.jpa.repository.RecallRepository
 import java.util.UUID
 
 @Service
 class RecallService(private val recallRepository: RecallRepository) {
   @Transactional
-  fun createRecall(createRecall: CreateRecall): CreateRecallResponse {
+  fun createRecall(createRecall: CreateRecall): SaveRecallResponse {
     val recall = recallRepository.save(RecallEntity.placeholderEntity(createRecall))
-    return CreateRecallResponse.from(recall)
+    return SaveRecallResponse.from(recall)
+  }
+
+  @Transactional
+  fun updateRecall(recallUniqueIdentifier: UUID, recall: CreateRecall): SaveRecallResponse {
+    val recallToUpdate = recallRepository.findOneByRecallUniqueIdentifier(recallUniqueIdentifier)
+      ?: recallRepository.save(RecallEntity.placeholderEntity(recall))
```

Comment:
```suggestion
      ?: recallRepository.save(RecallEntity.placeholderEntity(recall, recallUniqueIdentifier))
```
I think if you don't pass the UUID it will save it to a random UUID

Reply:
good spot!